src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -334,13 +334,14 @@ public PublicClientApplicationBuilder WithKerberosTicketClaim(string servicePrin
         /// On Mac, Linux and older versions of Windows a broker is not available.
         /// If your application is .NET5, please use the target .net5.0-windows10.0.17763.0 for all Windows versions and target net5.0 to target Linux and Mac.
         /// If your application is .NET classic or .NET Core 3.1 and you wish to use the Windows Broker, please install Microsoft.Identity.Client.Desktop first and call WithDesktopFeatures().
-        /// 
         /// On mobile, the device must be Intune joined and Authenticator or Company Portal must be installed. See https://aka.ms/msal-brokers
         /// </remarks>
+        [Obsolete("Use WithBroker by itself, which will fall back to using a web view if broker is unavailable.", false)]
         public bool IsBrokerAvailable()
-        {            
-            return PlatformProxyFactory.CreatePlatformProxy(null)
-                .CreateBroker(base.Config, null).IsBrokerInstalledAndInvokable(base.Config.Authority.AuthorityInfo.AuthorityType);
+        {
+            return Config.Authority?.AuthorityInfo?.AuthorityType == null ? false :

```

Comment:
How about we assume the authority type is AAD instead if it is not already configured?